src/event.rs

```diff
@@ -0,0 +1,566 @@
+		channel_id: [u8; 32],
+		/// The user_channel_id of the channel which is ready.
+		user_channel_id: u64,
+	},
+	/// A channel has been closed.
+	ChannelClosed {
+		/// The channel_id of the channel which has been closed.
+		channel_id: [u8; 32],
+		/// The user_channel_id of the channel which has been closed.
+		user_channel_id: u64,
+	},
+	// TODO: Implement on-chain events when better integrating with BDK wallet sync.
+	//OnChainPaymentSent {
+	//},
+	//OnChainPaymentReceived {
+	//}
+}
+
+// TODO: Figure out serialization more concretely - see issue #30
+impl Readable for LdkLiteEvent {
+	fn read<R: lightning::io::Read>(
+		reader: &mut R,
+	) -> Result<Self, lightning::ln::msgs::DecodeError> {
+		match Readable::read(reader)? {
+			0u8 => {
+				let payment_hash: PaymentHash = Readable::read(reader)?;
+				Ok(Self::PaymentSuccessful { payment_hash })
+			}
+			1u8 => {
+				let payment_hash: PaymentHash = Readable::read(reader)?;
+				Ok(Self::PaymentFailed { payment_hash })
+			}
+			2u8 => {
+				let payment_hash: PaymentHash = Readable::read(reader)?;
+				let amount_msat: u64 = Readable::read(reader)?;
+				Ok(Self::PaymentReceived { payment_hash, amount_msat })
+			}
+			3u8 => {
+				let channel_id: [u8; 32] = Readable::read(reader)?;
+				let user_channel_id: u64 = Readable::read(reader)?;
+				Ok(Self::ChannelReady { channel_id, user_channel_id })
+			}
+			4u8 => {
+				let channel_id: [u8; 32] = Readable::read(reader)?;
+				let user_channel_id: u64 = Readable::read(reader)?;
+				Ok(Self::ChannelClosed { channel_id, user_channel_id })
+			}
+			//5u8 => {
+			// TODO OnChainPaymentSent
+			//}
+			//6u8 => {
+			// TODO OnChainPaymentReceived
+			//}
+			_ => Err(lightning::ln::msgs::DecodeError::InvalidValue),
+		}
+	}
+}
+
+impl Writeable for LdkLiteEvent {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), lightning::io::Error> {
+		match self {
+			Self::PaymentSuccessful { payment_hash } => {
+				0u8.write(writer)?;
+				payment_hash.write(writer)?;
+				Ok(())
+			}
+			Self::PaymentFailed { payment_hash } => {
+				1u8.write(writer)?;
+				payment_hash.write(writer)?;
+				Ok(())
+			}
+			Self::PaymentReceived { payment_hash, amount_msat } => {
+				2u8.write(writer)?;
+				payment_hash.write(writer)?;
+				amount_msat.write(writer)?;
+				Ok(())
+			}
+			Self::ChannelReady { channel_id, user_channel_id } => {
+				3u8.write(writer)?;
+				channel_id.write(writer)?;
+				user_channel_id.write(writer)?;
+				Ok(())
+			}
+			Self::ChannelClosed { channel_id, user_channel_id } => {
+				4u8.write(writer)?;
+				channel_id.write(writer)?;
+				user_channel_id.write(writer)?;
+				Ok(())
+			} //Self::OnChainPaymentSent { .. } => {
+			  //TODO
+			  //}
+			  //Self::OnChainPaymentReceived { .. } => {
+			  //TODO
+			  //}
+		}
+	}
+}
+
+pub(crate) struct LdkLiteEventQueue<K: KVStorePersister> {
+	queue: Mutex<EventQueueSerWrapper>,
+	notifier: Condvar,
+	persister: Arc<K>,
```

Comment:
Should we use `Deref` or is `Arc` required for the runtime?

Reply:
Yeah so far I opted to use concrete types everywhere since it's not entirely clear what we'll want to make configurable. That said, the persister will surely be configurable, so I now made it `Deref` with https://github.com/lightningdevkit/ldk-lite/pull/10/commits/4240272a87a8dbf32d06ce6320d9c1aa7563294d.